src/main/ApplicationImpl.cpp

```diff
@@ -549,27 +549,109 @@ ApplicationImpl::joinAllThreads()
     LOG(DEBUG) << "Joined all " << mWorkerThreads.size() << " threads";
 }
 
-bool
-ApplicationImpl::manualClose()
+std::string
+ApplicationImpl::manualClose(optional<uint32_t> const& manualLedgerSeq,
+                             optional<TimePoint> const& manualCloseTime)
 {
     // Manual close only makes sense for validating nodes
     if (!mConfig.NODE_IS_VALIDATOR)
     {
-        return false;
+        throw std::invalid_argument(
+            "Issuing a manual ledger close requires NODE_IS_VALIDATOR to be "
+            "set to true.");
     }
 
     if (mConfig.MANUAL_CLOSE)
     {
-        mHerder->triggerNextLedger(mLedgerManager->getLastClosedLedgerNum() + 1,
-                                   true);
-        return true;
+        auto const startLedgerSeq = getLedgerManager().getLastClosedLedgerNum();
+        if (startLedgerSeq == std::numeric_limits<uint32_t>::max())
```

Comment:
Right, I noticed in testing that the `scphistory` database crashes on such values because it stores the ledger number as an `INT`.  I've updated the validation to reflect that.